src/GridLayoutProvider.ts

```diff
@@ -54,4 +56,24 @@ export class GridLayoutProvider extends LayoutProvider {
       throw new Error("setLayout called before layoutmanager was created, cannot be handled");
     }
   }
+
+  private setPreciseLayout(dimension: Dimension, index: number): void {
+    const maxSpan: number = this._maxSpan;
+    const itemSpan: number = this._getSpan(index);
+    if (itemSpan > maxSpan) {
+      throw new Error("Item span for index " + index + " is more than the max span");
+    }
+    if (this._renderWindowSize) {
+      if (this._isHorizontal) {
+        dimension.width = +(this._getHeightOrWidth(index).toFixed(2));
```

Comment:
pass the dimension to a custom function that reads precision variable and does toFixed? removal will be easier that way.

Reply:
updated